node/src/tasks/msp_move_bucket.rs

```diff
@@ -0,0 +1,306 @@
+                        Err(error) => {
+                            error!(
+                                target: LOG_TARGET,
+                                "Failed to decode file key proof for chunk {:?} of file {:?}: {:?}",
+                                chunk, file_key, error
+                            );
+                            continue;
+                        }
+                    };
+
+                    let proven = match file_key_proof.proven::<StorageProofsMerkleTrieLayout>() {
+                        Ok(chunk_data) => chunk_data,
+                        Err(error) => {
+                            error!(
+                                target: LOG_TARGET,
+                                "Failed to get proven data for file key proof: {:?}",
+                                error
+                            );
+                            continue;
+                        }
+                    };
+
+                    if proven.len() != 1 {
+                        error!(
+                            target: LOG_TARGET,
+                            "Expected exactly one proven chunk but got {}.",
+                            proven.len()
+                        );
+                        continue;
+                    }
+
+                    let chunk_data = proven[0].data.clone();
+                    let chunk_id = ChunkId::new(chunk);
+
+                    if chunk_id != proven[0].key {
+                        error!(
+                            target: LOG_TARGET,
+                            "Expected chunk id {:?} but got {:?}",
+                            chunk, proven[0].key
+                        );
+                        continue;
+                    }
+
+                    if let Err(error) = self
+                        .storage_hub_handler
+                        .file_storage
+                        .write()
+                        .await
+                        .write_chunk(&file_key, &chunk_id, &chunk_data)
+                    {
+                        // TODO: Handle this error better.
+                        error!(
+                            target: LOG_TARGET,
+                            "Failed to write chunk {:?} of file {:?} to storage: {:?}",
+                            chunk, file_key, error
+                        );
+                    } else {
+                        // We successfully downloaded the chunk, so we can break out of the retry loop.
+                        break;
+                    }
+                }
+            }
+
+            info!(
+                target: LOG_TARGET,
+                "MSP: inserting downloaded file {:?} of bucket {:?} to forest storage",
+                file_key,
+                event.bucket_id,
+            );
+
+            if let Err(error) = forest_storage
+                .write()
+                .await
+                .insert_files_metadata(&[file_metadata])
```

Comment:
We should check that we downloaded successfully downloaded and inserted all chunks in file storage before adding the file to our forest no?